apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -1,191 +1,204 @@
+    }
+    debug('render page');
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
 
-  if (req.body.__action === 'close') {
-    const assessment_id = res.locals.assessment.id;
-    const assessment_instance_id = req.body.assessment_instance_id;
-    checkBelongs(assessment_instance_id, assessment_id, (err) => {
-      if (ERR(err, next)) return;
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw error.make(403, 'Access denied (must be a student data editor)');
+    }
 
+    if (req.body.__action === 'close') {
+      const assessment_id = res.locals.assessment.id;
+      const assessment_instance_id = req.body.assessment_instance_id;
+      checkBelongsAsync(assessment_instance_id, assessment_id);
       const requireOpen = true;
       const close = true;
       const overrideGradeRate = true;
-      gradeAssessmentInstance(
+      gradeAssessmentInstanceAsync(
         assessment_instance_id,
         res.locals.authn_user.user_id,
         requireOpen,
         close,
         overrideGradeRate,
-        function (err) {
-          if (ERR(err, next)) return;
-          res.send(JSON.stringify({}));
-        },
       );
-    });
-  } else if (req.body.__action === 'delete') {
-    const assessment_id = res.locals.assessment.id;
-    const assessment_instance_id = req.body.assessment_instance_id;
-    checkBelongs(assessment_instance_id, assessment_id, (err) => {
-      if (ERR(err, next)) return;
-
+      res.send(JSON.stringify({}));
+    } else if (req.body.__action === 'delete') {
+      const assessment_id = res.locals.assessment.id;
+      const assessment_instance_id = req.body.assessment_instance_id;
+      checkBelongsAsync(assessment_instance_id, assessment_id);
       const params = [assessment_instance_id, res.locals.authn_user.user_id];
-      sqldb.call('assessment_instances_delete', params, function (err) {
-        if (ERR(err, next)) return;
-        res.send(JSON.stringify({}));
-      });
-    });
-  } else if (req.body.__action === 'grade_all' || req.body.__action === 'close_all') {
-    const assessment_id = res.locals.assessment.id;
-    const close = req.body.__action === 'close_all';
-    const overrideGradeRate = true;
-    util.callbackify(gradeAllAssessmentInstances)(
-      assessment_id,
-      res.locals.user.user_id,
-      res.locals.authn_user.user_id,
-      close,
-      overrideGradeRate,
-      function (err, job_sequence_id) {
-        if (ERR(err, next)) return;
-        res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
-      },
-    );
-  } else if (req.body.__action === 'delete_all') {
-    const params = [res.locals.assessment.id, res.locals.authn_user.user_id];
-    sqldb.call('assessment_instances_delete_all', params, function (err) {
-      if (ERR(err, next)) return;
+      sqldb.callAsync('assessment_instances_delete', params);
```

Comment:
This is an async function; it should be awaited.

Consider inlining the `params` array into that call.

Reply:
This is now being awaited in commit [8472ae8](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/8472ae8a0869182f87c71e15b748cbd6064db927). The `params` array has now been inlined as well.